test/AutoDiff/class_method.swift

```diff
@@ -0,0 +1,353 @@
+// RUN: %target-run-simple-swift
```

Comment:
Please also add a test where a custom VJP is defined in the parent class.

```swift
class A {
    func vjpFoo(...)
}

class B: A {
    @differentiable(vjp: vjpFoo)
    func foo(...)
}
```

Reply:
That is currently rejected by `@differentiable` attribute type-checking, because the VJP is defined in a different type context (superclass) than the original function (subclass).

```swift
class Super {
  @differentiable(vjp: vjpFoo)
  func foo(_ x: Float) -> Float {
    return x
  }
  final func vjpFoo(_ x: Float) -> (value: Float, pullback: (Float) -> Float) {
    return (foo(x), { v in v })
  }
}

class Sub : Super {
  @differentiable(vjp: vjpFoo)
  override func foo(_ x: Float) -> Float {
    return x * x
  }
}
```
```
tf-649.swift:12:24: error: 'vjpFoo' is not defined in the current type context
  @differentiable(vjp: vjpFoo)
                       ^
```

Eventually, we may want to support:
* For `@differentiable` attribute: referencing JVP/VJP from superclass.
* For `@differentiating` attribute: referencing original function from superclass.
* Overriding JVP/VJP without overriding original function.
```swift
class Super {
  @differentiable
  func foo(_ x: Float) -> Float {
    return x
  }
  @differentiating(foo)
  final func _(_ x: Float) -> (value: Float, pullback: (Float) -> Float) {
    return (foo(x), { v in v })
  }
}

class Sub : Super {
  // TF-649: Override JVP/VJP without overriding original function.
  @differentiating(foo)
  final func _(_ x: Float) -> (value: Float, pullback: (Float) -> Float) {
    return (foo(x), { v in v })
  }
}
```

[TF-649](https://bugs.swift.org/browse/TF-649) tracks this issue - please see it for more info.